packages/lib-user/src/components/shared/MainContent/MainContent.js

```diff
@@ -67,139 +46,187 @@ function MainContent({
   source = DEFAULT_SOURCE
 }) {
   const [activeTab, setActiveTab] = useState(0)
+  const [showCalendar, setShowCalendar] = useState(false)
+  const [customDateRange, setCustomDateRange] = useState([selectedDateRange.startDate, selectedDateRange.endDate])
+  
   const handleActiveTab = useCallback((tabIndex) => {
     setActiveTab(tabIndex)
   }, [])
 
+  useEffect(function updateCustomDateRange() {
+    setCustomDateRange([selectedDateRange.startDate, selectedDateRange.endDate])
+  }, [selectedDateRange])
+  
   const size = useContext(ResponsiveContext)
 
   const hoursSpent = convertStatsSecondsToHours(stats?.time_spent)
 
-  // create project options
-  let projectOptions = [
-    { label: 'ALL PROJECTS', value: 'AllProjects' },
-    ...(projects || []).map(project => ({
-      label: project.display_name,
-      value: project.id
-    }))
-  ]
-  const selectedProjectOption = projectOptions.find(option => option.value === selectedProject)
-
-  // create date range options
-  const dateRangeOptions = getDateRangeSelectOptions(source?.created_at?.substring(0, 10))
+  const { dateRangeOptions, selectedDateRangeOption } = getDateRangeSelectOptions({ created_at: source?.created_at?.substring(0, 10), selectedDateRange })
+  
+  const { projectOptions, selectedProjectOption } = getProjectSelectOptions({ projects, selectedProject })
+  
   const todayUTC = new Date().toISOString().substring(0, 10)
```

Comment:
The helper function could be used here too to show that they're tied together with the other datestrings being parsed.